### PR TITLE
light-client-cli: add command to fetch archive blocks and verify them

### PR DESCRIPTION
**Pull request by eranrund**
Small iteration on the light-client-cli tool to retrieve stored blocks from S3 and store them inside a protobuf, while also verifying fetched blocks.
This lets me take the protobuf and use it for testing in my go bindings endeavor.

### CODE REVIEW

File: light-client/cli/src/bin/main.rs [hunk 1]
```diff
@@ -39,67 +73,130 @@ fn main() {
...
+        trusted_validator_set_start_block,
+        historical_validator_sets: Default::default(),
+        known_valid_block_ids: Default::default(),
+    };
+
+    println!(
+        "{}",
+        serde_json::to_string_pretty(&light_client_verifier).unwrap()
+    );
+}
```
Comment (eranrund) [inline light-client/cli/src/bin/main.rs:157]: All of this just moved from `main()` into this function.

**Review by cbeck88 — APPROVED** [review]

File: light-client/cli/src/bin/main.rs [hunk 2]
```diff
@@ -2,27 +2,61 @@
...
+        tx_source_urls: Vec<String>,
+
+        /// Block index we are interested in.
+        #[clap(long, env = "MC_BLOCK_INDEX")]
+        block_index: BlockIndex,
+
+        /// File to write the fetched ArchiveBlocks protobuf to.
+        #[clap(long, env = "MC_OUT_FILE")]
+        out_file: PathBuf,
+
```
Comment (nick-mobilecoin) [inline light-client/cli/src/bin/main.rs:54]: :upside_down_face: This behavior of requiring an output file seems inconsistent with the `GenerateConfig` command.
One writes JSON to stdout, another requires a file.

Has something like https://docs.rs/clio/latest/clio/ been considered to make these CLI's a bit more consistent?

File: light-client/cli/src/bin/main.rs [hunk 3]
```diff
@@ -39,67 +73,130 @@ fn main() {
...
+                .expect("failed fetching block data")
+        })
+        .collect::<Vec<_>>();
+
+    if let Some(path) = light_client_verifier_config_path {
+        let json_data =
+            fs::read_to_string(path).expect("failed reading LightClientVerifierConfig file");
+        let light_client_verifier_config: LightClientVerifierConfig =
+            serde_json::from_str(&json_data).expect("failed parsing LightClientVerifierConfig");
+        let light_client_verifer = LightClientVerifier::from(light_client_verifier_config);
```
Comment (nick-mobilecoin) [inline light-client/cli/src/bin/main.rs:182]: :upside_down_face: nit
```suggestion
        let light_client_verifier = LightClientVerifier::from(light_client_verifier_config);
```

File: light-client/verifier/src/verifier.rs [hunk 4]
```diff
@@ -114,6 +114,37 @@ impl LightClientVerifier {
...
+        {
+            return Err(Error::BlockDataMismatch);
+        }
+
+        let block = block_data[0].block();
+        let block_contents = block_data[0].contents();
+
+        let block_metadata = block_data
+            .iter()
+            .filter_map(|block_data| block_data.metadata().cloned())
```
Comment (nick-mobilecoin) [inline light-client/verifier/src/verifier.rs:142]: :+1: That's a neat way to normalize the Optional metadata.

**Review by nick-mobilecoin — APPROVED** [review]
I don't think it's necessarily a blocker, but the `GenerateConfig` and `FetchArchiveBlocks` behavior deviate with respect to the handling of the output(file)

File: light-client/cli/src/bin/main.rs (hunk 2 above)
Comment (eranrund) [reply light-client/cli/src/bin/main.rs:54]: No, but that looks like a nice crate so I'll switch to that.

**Pull request merged**